core/debug/src/debug.c

```diff
@@ -230,3 +230,117 @@ void debug_fault(THaltError reason, uint32_t lr, uint32_t sp)
     debug_mpu_config();
     DEBUG_PRINT_END();
 }
+
+static void __debug_reboot(void)
+{
+    UVISOR_SVC(UVISOR_SVC_ID_DEBUG_REBOOT, "");
+}
+
+void debug_reboot(void)
+{
+    if (!g_debug_box.initialized || g_active_box != g_debug_box.box_id) {
+        HALT_ERROR(NOT_ALLOWED, "This function can only be called from the context of an initialized debug box.\n\r");
+    }
+
+    /* Reboot.
+     * This can only be called from privileged code. */
+    NVIC_SystemReset();
+}
+
+/* FIXME: Currently it is not possible to return to a regular execution flow
+ *        after the execution of the debug box handler. It is possible to
+ *        reboot, though. */
```

Comment:
It's possible to return but it requires to develop the context switch *out*. I'd take this shortcut for the moment so that we have a smaller test surface. It's not trivial to perform the switch out, especially if we must take into consideration de-privileging from a fault.